monitor/verifier.go

```diff
@@ -237,53 +235,59 @@ func (m *Monitor) verifyRequest(change sync.Change) (bool, bool) {
 	m.operator.CliCtx.Codec.MustUnmarshalBinaryBare(change.Data, &request)
 	logmsg := fmt.Sprintf("verify change id %d, request: %s", change.ID, request)
 
-	var signedSimplexState chain.SignedSimplexState
-	err := proto.Unmarshal(request.SignedSimplexStateBytes, &signedSimplexState)
+	signedSimplexState, simplexChannel, err := common.UnmarshalSignedSimplexStateBytes(request.SignedSimplexStateBytes)
 	if err != nil {
 		log.Errorf("%s. unmarshal signedSimplexStateBytes err: %s", logmsg, err)
-		return false, false
+		return true, false
 	}
 
-	r, err := subscribe.GetRequest(m.operator.CliCtx, m.ethClient.Ledger, &signedSimplexState)
-	if err != nil {
-		log.Errorf("%s. get request err: %s", logmsg, err)
-		return false, false
+	_, err = m.getRequest(simplexChannel.ChannelId, mainchain.Bytes2Hex(simplexChannel.PeerFrom))
+	if err == nil {
```

Comment:
`err != nil` does not mean that request is not initiated

Reply:
right, what is the correct way to tell if a key exists here?

Reply:
fixed